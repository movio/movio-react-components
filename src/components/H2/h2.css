@import "variables.css";

:root {
  --h2-font-family: var(--font-primary);
  --h2-font-size: 1.714rem;
  --h2-color: var(--color-neutral-2);
}

.h2 {
  font-family: var(--h2-font-family);
  font-size: var(--h2-font-size);
  color: var(--h2-color);
  font-weight: 300;
}
