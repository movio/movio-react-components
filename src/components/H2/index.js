// @flow
import React from 'react';
import classnames from 'classnames';

import styles from './h2.css';

type Props = {
  children?: ReactChildren,
  className?: string,
};

const H2 = ({ className, children }: Props) => (
  <h2 className={classnames(className, styles.h2)}>
    {children}
  </h2>
);

export default H2;
export { styles };
