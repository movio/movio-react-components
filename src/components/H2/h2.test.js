import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import H2, { styles } from './index';

describe('<H2 />', () => {
  it('should correctly render children, when text', () => {
    const component = <H2>Title</H2>;
    const wrapper = shallow(component);
    expect(wrapper.text()).toEqual('Title');
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h2-children-text');
  });

  it('should append a className to the default, if provided', () => {
    const component = <H2 className="test-class">Title H2</H2>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(`test-class ${styles.h2}`);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h2-custom-classname');
  });

  it('should use the default className from the stylesheet, if not provided', () => {
    const component = <H2>Title H2</H2>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(styles.h2);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h2-default-classname');
  });
});
