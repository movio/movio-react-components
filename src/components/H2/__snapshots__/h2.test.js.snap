// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h2-children-text 1`] = `
<h2
  className="h2"
>
  Title
</h2>
`;

exports[`h2-custom-classname 1`] = `
<h2
  className="test-class h2"
>
  Title H2
</h2>
`;

exports[`h2-default-classname 1`] = `
<h2
  className="h2"
>
  Title H2
</h2>
`;
